From db34938d14084a6f57379b6f7e55018e0ab64719 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:26:36 -0400
Subject: [tools] perf/inject: Do not repipe attributes to a perf.data file

Message-id: <1408462094-14194-228-git-send-email-jolsa@redhat.com>
Patchwork-id: 88201
O-Subject: [PATCH RHEL7.1 BZ1131394 227/325] perf inject: Do not repipe attributes to a perf.data file
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 89c97d936e76b064a52ee056602b2a62b3f1ef70
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Tue Oct 22 10:34:09 2013 +0300

description
===========
perf.data files contain the attributes separately, do not put them in
the event stream as well.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-inject.c b/tools/perf/builtin-inject.c
index eb1a594..409ceaf 100644
--- a/tools/perf/builtin-inject.c
+++ b/tools/perf/builtin-inject.c
@@ -72,12 +72,17 @@ static int perf_event__repipe_attr(struct perf_tool *tool,
        union perf_event *event,
        struct perf_evlist **pevlist)
 {
+ struct perf_inject *inject = container_of(tool, struct perf_inject,
+        tool);
  int ret;
 
  ret = perf_event__process_attr(tool, event, pevlist);
  if (ret)
   return ret;
 
+ if (!inject->pipe_output)
+  return 0;
+
  return perf_event__repipe_synth(tool, event);
 }
 
-- 
1.7.1